src/gsy_e/models/market/forward.py

```diff
@@ -31,6 +32,15 @@
     from gsy_e.models.area.event_dispatcher import AreaDispatcher
 
 
+@dataclass(frozen=True)
+class ForwardMarketSlotParameters:
+    """Parameters that describe a forward market slot."""
+    open_timestamp: DateTime
+    close_timestamp: DateTime
+    delivery_start_timestamp: DateTime
+    delivery_end_timestamp: DateTime
```

Comment:
A few points about this. First just a small nit: I think that there's a difference between timestamp and datetime. To mitigate this maybe we could just use something like:
```suggestion
    opening_time: DateTime
    closing_time: DateTime
    delivery_start_time: DateTime
    delivery_end_time: DateTime
```
(time is also a bit different, but maybe it's more generic).

Reply:
Other two more relevant (IMO) points:
1. The "delivery times" are basically the same thing as just `market.time_slot`. What I mean is that, in the case of a yearly market (for example), the slot would be `2024-01-01T00:00:00` and it would already represent on its own the entire delivery time (because its duration is 1 year).
I find this a bit confusing: the market's time slot represents a **period** of time, while the delivery start (or end) time represents a precise **point** in time. IMO this difference is not immediately evident from the code now and it might lead to issues and confusion if we introduce both concepts. One way to make this more obvious IMO would be by creating a specific primitive class for market slots.
2. Since these are the parameters of a market slot, why don't we create a `MarketSlot` class and add these parameters inside of it? This would remove the need for all the `get_market_parameters_for_market_slot` logic, because each market slot would already have its parameters inside of it. I know this would take some time to implement (and it wouldn't be in the context of these PRs), but I think it would help (also given what I said in point 1). We could use slots if you're worried about performance. What do you think?

Reply:
Some remarks on the points:
1. Actually the time_slot (the way we use it at least) is just a timestamp that identifies the market. If you just check the type of the time_slot it is a DateTime and not a Duration. In order to represent the duration aspect of this, each market has to implement this method `_get_market_slot_duration` which is indeed representing the length of the delivery period. We could indeed wrap all these in a separate class, however I am afraid of the changes that we will have to make to our legacy code, and I personally think it is not worth doing this for the old markets at the very end, since the method I mentioned makes its intent quite clear. Having said that, let me try to see if it is feasible to implement something like you mention in point #2 only for the forward markets, where we will have more use of it.

Reply:
Ah and of course I am fine with the nits. After all this is a draft and I wanted your opinion on the naming too, thanks